guide.md

```diff
@@ -0,0 +1,440 @@
+# A guide to rules_nixpkgs
+
+TODO: is the flake shell pure? because if this is run on a non-nixos system, and /usr/bin/gcc continues to be visible, that messes things up
+TODO: document nixpkgs.nix
+
+---
+
+Bazel is a modern build system valued for performant, correct, and hermetic
+builds. It has a few shortcomings, one of which is that that Bazel is only
+hermetic if certain conditions are met, and they aren't in many people's build
+setups. In practice, Bazel build products fail to be reproducible because Bazel
+silently depends on various things present in the global environment, like a
+C++ compiler. This ties into another issue, which is that external dependencies
```

Comment:
Perhaps the external deps thing should be its own paragraph. Yes, it's related, and rules_nixpkgs covers both, but it's still a somewhat separate concern.

Reply:
Done